docs/DevGuide/Parallelization.md

```diff
@@ -0,0 +1,539 @@
+      memory parallelism) builds, a node corresponds to the usual definition of
+      a node on a supercomputer. However, for non-SMP builds nodes and cores are
+      equivalent. We ensure that all entry method calls done through the
+      Algorithm's `simple_action` and `receive_data` functions are
+      threadsafe. User controlled threading is possible by calling the non-entry
+      method member function `threaded_action`.
+2. `using metavariables` is set to the Metavariables struct that stores the
+   global metavariables. It is often easiest to have the %Parallel
+   Component struct have a template parameter `Metavariables` that is the
+   global metavariables struct. Examples of this technique are given below.
+3. `using phase_dependent_action_list` is set to a `tmpl::list` of
+   `Parallel::PhaseActions<PhaseType, Phase, tmpl::list<Actions...>>` where each
+   `PhaseAction` represents a PDAL that will be executed on the parallel
+   component during the specified phase. The  %Actions are executed in the order
+   that they are given in the `tmpl::list`s of the PDALs, but the phases need
+   not be run in linear order. However, `DataBox` types are constructed assuming
+   the phases are performed from first in the `phase_dependent_action_list` to
+   the last. Simple actions (described below) can be executed in any phase. If
+   there are no iterable actions in a phase then a `PhaseAction` need not be
+   specified for that phase. However, at least one `PhaseAction`, even it if is
+   empty, must be specified.
+4. `using options` is set to a (possibly empty) `tmpl::list` of the option
+   structs. The options are read in from the input file specified in the main
+   `Metavariables` struct. After being read in they are passed to the
+   `initialize` function of the parallel component, which is described below.
+5. `using add_options_to_databox` is set to a struct that contains a static
+   `apply` function taking arguments `(db::DataBox<DbTagsList>&&, options...)`
+   where the `options...` are the arguments read from the input file using the
+   `options` type alias. Typically the `add_options_to_databox` will insert the
+   options into the `DataBox` for use during initialization. The
+   `add_options_to_databox` struct must also contain a type list named
+   `simple_tags` which contains a list of all the tags that the `apply` function
+   will add to the DataBox. No compute tags may be added at this stage. A
+   reasonable location for this struct is inside the action that will do the
+   initialization. Two reasonable defaults are provided,
+   `Parallel::AddNoOptionsToDataBox` and
+   `Parallel::ForwardAllOptionsToDataBox`.
+6. `using const_global_cache_tag_list` is set to a `tmpl::list` of OptionTags
+   that are required by the parallel component. This is usually obtained from
+   the `action_list` using the `Parallel::get_const_global_cache_tags`
+   metafunction.
```

Comment:
suggestion:  These options lead to items that are stored in the ConstGlobalCaches (of which there is one copy per Charm++ node)